### DISCUSSION TITLE
Installing Nextcloud AIO on TrueNAS SCALE

### DISCUSSION THREAD

**Discussion by ghost** [nextcloud/all-in-one#1200] [Questions] [answered]
Can anyone help me with this?

I do not know what I need to add to environment variables, volume, ...

Possibly interesting for the documentation???

**Comment by ghost** [comment]
I don't understand what I have to enter in Hosts and Variables. It would be very nice if someone could help me.
Thank you in advance :)

![grafik](https://user-images.githubusercontent.com/90777102/192624548-c0e6be92-71a4-442d-a795-d62ac63c93b7.png)
![grafik](https://user-images.githubusercontent.com/90777102/192624815-a9ad9a3f-215a-4762-ae47-688c8409d08f.png)

**Comment by szaimen** [comment]
I guess easiest is to connect to your instance via cli: https://www.truenas.com/docs/scale/gettingstarted/install/consolesetupmenuscale/ and then run the docker run command of the readme. However this is not tested and you probably need to follow the [reverse proxy documentation](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md)

**Reply by ghost** [reply]
I will test it... Is there still a way to create the whole thing with the GUI?

**Reply by szaimen** [reply]
Most likely yes but I find the GUI of Truenas as cryptic as you.

**Comment by szaimen** [comment, marked as answer]
See https://github.com/nextcloud/all-in-one#can-i-run-aio-on-truenas-scale

**Reply by ghost** [reply]
Since IX System dumped Kubernetes and fully integrated Docker (compose) with 24.10 (Electric Eel) which is now a stable release - this should be revisited.

**Reply by szaimen** [reply]
Really? Wow, cool! Would you mind submitting a PR? :)

**Reply by ghost** [reply]
Hi Simon, a PR for what ?

I migrated my AIO instance to Truenas Scale 24.10.0 (Eletric Eel) by taking the following steps:

- Accessing AIO Mastercontainer Webinterface on the OLD instance.
- Stop all Containers
- Perform Borg Backup of old AIO Instance
- On Scale: Install "dockge" from Apps Menu - This basically allows you to run your docker stacks easily on Scale (Portainer is also an option for the people more familiar to this)
My advice here is to create a dataset for your docker stacks (e.g. /mnt/tank/docker)
Then point dockge to /mnt/tank/docker and its datadir to /mnt/tank/docker/dockge

- Deploy and start AIO container on Truenas Scale via console or dockge - Important to create your Nextcloud Data Dataset on Scale  and point the Mastercontainer to your new Data-Dir with the full path on SCALE (e.g. /mnt/tank/docker/nextcloud-aio/)
- Start AIO Mastercontainer on the new system
- Restore Backup from the old instance
- Start all the containers

Since I had my data Directory already running on the NAS (which was Truenas Core before) this process was quiet straight forward.
I just needed to adjust the path in AIO Mastercontainer Compose File and point it to the correct path for data directory.

Hope this helps - Please let me know if you need any further information.

**Reply by szaimen** [reply]
> Hi Simon, a PR for what ?

A PR to adjust this documentation: https://github.com/nextcloud/all-in-one#can-i-run-aio-on-truenas-scale

**Reply by ghost** [reply]
Done

**Reply by szaimen** [reply]
Thanks! Having a look :)
